### PR TITLE
Add the prometheus-collector as a new module

### PR DESCRIPTION
**Pull request by jkwatson**
**Description:**

Adds a new module for a prometheus collector. This was taken from the opentelemetry-java core repository originally.

**Testing:**
Unit tests were moved over as well.

**Documentation:**

Added a basic README.

### CODE REVIEW

File: prometheus-collector/src/main/java/io/opentelemetry/contrib/prometheus/PrometheusCollector.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    this.collector.register();
+  }
+
+  /**
+   * Returns a new {@link PrometheusCollector} to be registered with a {@link
+   * io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder}.
+   */
+  public static PrometheusCollector create() {
+    return new PrometheusCollector();
+  }
```
Comment (jack-berg) [inline prometheus-collector/src/main/java/io/opentelemetry/contrib/prometheus/PrometheusCollector.java:44]: Could be worthwhile to accept a `CollectorRegistry` argument to this to allow registering to registries other than `CollectorRegistry.defaultRegistry`.

**Review by jack-berg — APPROVED** [review]

File: prometheus-collector/src/main/java/io/opentelemetry/contrib/prometheus/PrometheusCollector.java (hunk 1 above)
Comment (jkwatson) [reply prometheus-collector/src/main/java/io/opentelemetry/contrib/prometheus/PrometheusCollector.java:44]: sounds like a feature request!

Comment (jack-berg) [reply prometheus-collector/src/main/java/io/opentelemetry/contrib/prometheus/PrometheusCollector.java:44]: And not something I need myself so should wait

**Review by mateuszrzeszutek — APPROVED** [review]

File: prometheus-collector/src/main/java/io/opentelemetry/contrib/prometheus/NameSanitizer.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+package io.opentelemetry.contrib.prometheus;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+/** Sanitizes a metric or label name. */
+class NameSanitizer implements Function<String, String> {
```
[the remark this answers is not in the document]
Comment (jkwatson) [reply prometheus-collector/src/main/java/io/opentelemetry/contrib/prometheus/NameSanitizer.java:14]: Before changing anything in this module, I'd like to get it over here, as-is from the core repo, then we can start cleaning it up/enhancing/etc if that's ok.

File: prometheus-collector/src/test/java/io/opentelemetry/contrib/prometheus/PrometheusCollectorTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+// Tests deprecated class
```
[the remark this answers is not in the document]
Comment (jkwatson) [reply prometheus-collector/src/test/java/io/opentelemetry/contrib/prometheus/PrometheusCollectorTest.java]: fixed

File: prometheus-collector/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+## Usage
+
+```
+sdkMeterProvider.registerMetricReader(PrometheusCollector.create());
+```
+
+## Component owners
+
+- [John Watson](https://github.com/jkwatson), Verta.ai
```
[the remark this answers is not in the document]
Comment (jkwatson) [reply prometheus-collector/README.md:17]: done

**Pull request merged**